build-info-extractor/src/main/java/org/jfrog/build/extractor/usageReport/UsageReporter.java

```diff
@@ -1,40 +1,63 @@
 package org.jfrog.build.extractor.usageReport;
 
+import org.apache.commons.codec.digest.DigestUtils;
 import org.jfrog.build.api.util.Log;
 import org.jfrog.build.client.ProxyConfiguration;
 import org.jfrog.build.extractor.clientConfiguration.client.artifactory.ArtifactoryManager;
 
 import java.io.IOException;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Enumeration;
 
 /**
  * Created by Bar Belity on 20/11/2019.
  */
 public class UsageReporter {
-    private String productId;
+    private final String productId;
     private FeatureId[] features;
+    private String uniqueClientId;
 
     public UsageReporter(String productId, String[] featureIds) {
         this.productId = productId;
         setFeatures(featureIds);
     }
 
+    private String generateUniqueClientId() throws SocketException {
```

Comment:
Let's add some JavaDoc to this function. Let's mention that the original MAC address is impossible to retrieve because we are using sha1 hashing.